Content.Client/UserInterface/Controls/RadialMenu.cs

```diff
@@ -205,51 +278,308 @@ private void OnClicked(ButtonEventArgs args)
+            // 0->2pi, so remap the result into that range
+            angle = MathF.PI * 2 + angle;
+        }
+
+        var isInAngle = angle >= _angleSectorFrom && angle < _angleSectorTo;
+        return isInAngle;
+    }
+
+    /// <summary>
+    /// Draw segment between two concentrated circles from and to certain angles.
+    /// </summary>
+    /// <param name="drawingHandleScreen">Drawing handle, to which rendering should be delegated.</param>
+    /// <param name="center">Point where circle center should be.</param>
+    /// <param name="radiusInner">Radius of internal circle.</param>
+    /// <param name="radiusOuter">Radius of external circle.</param>
+    /// <param name="angleSectorFrom">Angle in radian, from which sector should start.</param>
+    /// <param name="angleSectorTo">Angle in radian, from which sector should start.</param>
+    /// <param name="color">Color for drawing.</param>
+    /// <param name="filled">Should figure be filled, or have only border.</param>
+    private void DrawAnnulusSector(
+        DrawingHandleScreen drawingHandleScreen,
+        Vector2 center,
+        float radiusInner,
+        float radiusOuter,
+        float angleSectorFrom,
+        float angleSectorTo,
+        Color color,
+        bool filled = true
+    )
+    {
+        const float minimalSegmentSize = MathF.Tau / 32;
+
+        var requestedSegmentSize = angleSectorTo - angleSectorFrom;
+        var segmentCount = (int)(requestedSegmentSize / minimalSegmentSize) + 1;
+        var anglePerSegment = requestedSegmentSize / (segmentCount - 1);
+
+        var bufferSize = segmentCount * 2;
+        if (_sectorPointsForDrawing == null || _sectorPointsForDrawing.Length != bufferSize)
+        {
+            _sectorPointsForDrawing ??= new Vector2[bufferSize];
+        }
+
+        for (var i = 0; i < segmentCount; i++)
+        {
+            var angle = angleSectorFrom + anglePerSegment * i;
+
+            // Flip Y to get from ui coordinates to natural coordinates
+            var unitPos = new Vector2(MathF.Cos(angle), -MathF.Sin(angle));
+            var outerPoint = center + unitPos * radiusOuter;
+            var innerPoint = center + unitPos * radiusInner;
+            if (filled)
+            {
+                // to make filled sector we need to create strip from triangles
+                _sectorPointsForDrawing[i * 2] = outerPoint;
+                _sectorPointsForDrawing[i * 2 + 1] = innerPoint;
+            }
+            else
+            {
+                // to make border of sector we need points ordered as sequences on radius
+                _sectorPointsForDrawing[i] = outerPoint;
+                _sectorPointsForDrawing[bufferSize - 1 - i] = innerPoint;
+            }
+        }
+
+        var type = filled
+            ? DrawPrimitiveTopology.TriangleStrip
+            : DrawPrimitiveTopology.LineStrip;
+        drawingHandleScreen.DrawPrimitives(type, _sectorPointsForDrawing, color);
```

Comment:
having to call it on each draw is a bit wierd. But ok. fixed.

Reply:
If it's too expensive here, move it else where,or use a lookup table to convert between rgb and srgb colors

Reply:
Its not like 'too expensive', but we shove everything that sure we can be cached to the cache for perf. Not nice thing is that it will require some wierd getter-setter dancing. I don't mind it, gonna do it...

Reply:
done